library/debug.c

```diff
@@ -396,19 +396,22 @@ void mbedtls_debug_print_crt(const mbedtls_ssl_context *ssl, int level,
         return;
     }
 
+    buf = mbedtls_calloc(1, 1024);
+    if (buf == NULL) {
```

Comment:
I'm not sure this is acceptable (I welcome other opinions). This is silently hiding logs in low memory conditions. Those logs are mostly intended as debug logs, but they _can_ be used to keep tracks of the activity on a server. A malicious client could saturate a server with connections, causing heap exhaustion, and then exploit some bug without being logged.

Reply:
ok, different priorities, i guess... in the land where i come from, 1.5K of stack is considered a luxury :)
our entire task has 4K stack - and mbedtls still works, handshake and all. however, it blew up when i turned on the debug. such high usage for a single function seems a bit excessive to me.